packages/react-testing/src/tests/e2e.test.tsx

```diff
@@ -173,31 +173,34 @@ describe('@shopify/react-testing', () => {
       );
     }
 
-    it('releases any stale promises when component is destroyed', async () => {
-      const wrapper = mount(<Counter />);
-      wrapper.act(() => new Promise(() => {}));
-      wrapper.act(() => new Promise(() => {}));
-      await wrapper.destroy();
+    // eslint-disable-next-line no-process-env
+    if (process.env.REACT_VERSION !== '17') {
```

Comment:
This seem reasonable although I don't know if anyone is aware of a better way of achieving this as I didn't see any precent of the use of `process.env.REACT_VERSION` in the codebase.

Reply:
Rando idea to avoid nesting adjustments: 

```ts
const itExceptReact17 = process.env.REACT_VERSION !== '17' ? it : it.skip;

itExceptReact17('releases any stale promises when component is destroyed', async () => {
```

If the eventual plan is to remove the "stale promise releasing" behaviour in 17, and make it a react 18 only feature, as the behaviour in v17 is not stable for unknown reasons then I think I'd prefer it if we roll this change and that disabling in the same PR rather than have two separate PRs.

Reply:
Agreed, better to still have the `it` block but have it be skipped (shows up in test reporters that way)
Alternative approach (either is fine with me, I like this idea from the original [feature request](https://github.com/facebook/jest/issues/3652)):

```js
const itIf = condition => condition ? it : it.skip;

itIf(process.env.REACT_VERSION !== '17')('blah blah', () => {
  ...
});
```

Reply:
Sounds good @ryanwilsonperkin  , this PR was mostly meant to unblock tests while I work a fix for 17 queue emptying or disabling it all together. I had sometime this morning and I might have found a fix that I'll put up later.